nativelib/src/main/scala/scala/scalanative/native/Word.scala

```diff
@@ -0,0 +1,43 @@
+package scala.scalanative
+package native
+
+import scala.runtime.BoxesRunTime._
+import scala.reflect.ClassTag
+import scalanative.runtime._
+import scalanative.runtime.Intrinsics._
+import scalanative.runtime.Boxes._
+
+final class Word(private[scalanative] val rawWord: RawWord) {
+  @alwaysinline def toInt: Int     = castRawWordToInt(rawWord)
+  @alwaysinline def toUInt: UInt   = castRawWordToInt(rawWord).toUInt
+  @alwaysinline def toUWord: UWord = new UWord(rawWord)
+
+  @alwaysinline def +(other: Word): Word =
+    new Word(addRawWords(rawWord, other.rawWord))
+  @alwaysinline def -(other: Word): Word =
+    new Word(subRawWords(rawWord, other.rawWord))
+  @alwaysinline def *(other: Word): Word =
+    new Word(multRawWords(rawWord, other.rawWord))
+  @alwaysinline def /(other: Word): Word =
+    new Word(divRawWords(rawWord, other.rawWord))
```

Comment:
Lets have all ops that you can do on Longs as well. The ones missing are: unary ops (~, !, +, -), comparisons (<, <=, >, >=), bitwise ops (&, |, ^).